src/deque.py

```diff
@@ -0,0 +1,67 @@
+"""Implementation of Deque."""
+
+from dll import DoublyLinkedList
+
+
+class Deque(object):
+    """Class implementation of queue.
+
+    append(val): adds value to the end of the deque.
+    appendleft(val): adds a value to the front of the deque.
+    pop(): removes a value from the end of the deque and returns it
+            (raises an exception if the deque is empty)
+    popleft(): removes a value from the front of the deque and returns it
+            (raises an exception if the deque is empty)
+    peek(): returns the next value that would be returned by pop but
+            leaves the value in the deque (returns None if the deque is empty)
+    peekleft(): returns the next value that would be returned by popleft
+            but leaves the value in the deque (returns None if the deque
+            is empty)
+    size(): returns the count of items in the queue (returns 0 if the
+            queue is empty)
+
+    """
+
+    def __init__(self, iterable=None):
+        """Instatiate Queue."""
+        self.dll = DoublyLinkedList(iterable)
+
+    def append(self, contents):
+        """Add value to the end of the deque."""
+        self.dll.append(contents)
```

Comment:
Either head or tail should have a next node, the other should have a prev.  Neither should throw an error once at least one item has been added (should instead should return none).